src/main/java/com/epam/edumanagementsystem/admin/rest/api/AcademicClassController.java

```diff
@@ -37,64 +36,45 @@ public class AcademicClassController {
     private final AcademicClassService academicClassService;
     private final AcademicCourseService academicCourseService;
     private final StudentService studentService;
-    private final TimetableService timetableService;
     private final TeacherService teacherService;
 
     @Autowired
-    public AcademicClassController(AcademicClassService academicClassService, AcademicCourseService academicCourseService, StudentService studentService, TimetableService timetableService, TeacherService teacherService) {
+    public AcademicClassController(AcademicClassService academicClassService, AcademicCourseService academicCourseService, StudentService studentService, TeacherService teacherService) {
         this.academicClassService = academicClassService;
         this.academicCourseService = academicCourseService;
         this.studentService = studentService;
-        this.timetableService = timetableService;
         this.teacherService = teacherService;
     }
 
     @GetMapping
     @Operation(summary = "Shows academic class section on admin's dashboard with existing courses")
-    public String openAcademicCourse(Model model) {
-
-        List<AcademicClassDto> academicClassDtoList = academicClassService.findAll();
-        model.addAttribute("academicClasses", academicClassDtoList);
-        model.addAttribute("academicClass", new AcademicClassDto());
-        return "academicClassSection";
+    public String openAcademicClassSection(Model model) {
+        setAttributesInClassesSection(model);
+        return ACADEMIC_CLASSES_SECTION;
     }
 
     @PostMapping
     @Operation(summary = "Saves the created academic class")
-    public String create(@ModelAttribute("academicClass") @Valid AcademicClass academicClass,
-                         BindingResult result, Model model) {
-
-        List<AcademicClassDto> academicClassDtoList = academicClassService.findAll();
-        List<AcademicClass> academicClassList = AcademicClassMapper.academicClassessList(academicClassDtoList);
-        model.addAttribute("academicClasses", academicClassDtoList);
-
-        if (!result.hasFieldErrors("classNumber")) {
-            if (InputFieldsValidation.validateInputFieldSize(academicClass.getClassNumber())) {
-                model.addAttribute("nameSize", "Symbols can't be more than 50");
-            }
-            if (InputFieldsValidation.checkingForIllegalCharacters(academicClass
-                    .getClassNumber(), model)) {
-                model.addAttribute("invalidURL", "<>-_`*,:|() symbols can be used.");
-            }
-        }
+    public String save(@ModelAttribute("academicClass") @Valid AcademicClassDto academicClassDto,
+                       BindingResult result, Model model) {
 
+        List<AcademicClassDto> academicClassDtoList = getAllClassesAndSetAttributes(model);
+        validateNameField(model, academicClassDto, result);
```

Comment:
getAllClassesAndSetAttributes(model) could be passed as a param to validate nameField().  And  maybe here you could use a custom constraint  annotation for the field with validation all this validation logic